network/p2p/client.go

```diff
@@ -41,6 +41,7 @@ type Client struct {
 	handlerPrefix []byte
```

Comment:
Not part of this PR but should we have a `NewClient` function? If this in intended to be used only inside this package, `Client` can be unexported. If it's intended to be used outside this package, then a user can't use `Client` because they can't set its inner fields.

Reply:
This is created via `RegisterAppProtocol`. It's not possible to use a `Client` without a corresponding `Router` because the `Router` holds the map of pending requests -> the callbacks that need to be fired on response. We could theoretically refactor this into a `NewClient(Router, handlerID, ....)` if we felt that looked cleaner.

Reply:
It just feels weird to export this if it's actually unusable outside this package

Reply:
`Client` is usable outside of this package, `RegisterAppProtocol` initializes it [here](https://github.com/ava-labs/avalanchego/blob/499a7f2334166919addd7ce7b00d59c92dfba6d4/network/p2p/router.go#L72).

Maybe we could rename this to `NewClient` to be more verbose or remove the receiver on `Router`. This definitely might seem confusing coming from the existing paradigm of how we've developed VMs without the SDK component.

Reply:
Ah I see. Yeah I think it might be a bit confusing to someone looking at this package how to create a new client. I think maybe even just a comment on `Client` might suffice.

Reply:
Yeah I could see an argument to make the constructor code `NewClient(sender common.AppSender, router *Router, handlerID byte) *Client`. This would have the added benefit of not having to keep around the `sender` as a field in `Router` which we currently do.